src/Routing/UrlGenerator.php

```diff
@@ -142,13 +158,13 @@ function ($matches) use ($alias, &$parameters, $autoItems, &$hasAutoItem) {
                 $value = $parameters[$param];
                 unset($parameters[$param]);
 
-                if (!$hasAutoItem && in_array($param, $autoItems, true)) {
-                    $hasAutoItem = true;
-
-                    return $value;
+                if (!$config->get('useAutoItem') || $hasAutoItem || !in_array($param, $autoItems, true)) {
```

Comment:
Are you sure this is correct? The auto item should only be used if the config flag is true